Assets/Scripts/WeaponSystem/MockPlayer.cs

```diff
@@ -0,0 +1,18 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class MockPlayer : MonoBehaviour
+{
+    public GameObject EquippedWeapon;
+
+    private void Update()
+    {
+        if (Input.GetKeyDown(KeyCode.A))
+            EquippedWeapon.GetComponent<Weapon>().TryDoLightAttack();
```

Comment:
Instead of calling GetComponent every frame, you should cache the reference to the weapon in a field.
private Weapon _weapon

when from the OnAwake method you can call the GetComponent method. By doing it this way we only need to call the GetComponent method once. :)